rust/processor/src/gap_detectors/mod.rs

```diff
@@ -28,9 +28,17 @@ pub enum GapDetector {
 
 #[enum_dispatch(GapDetectorTrait)]
 pub enum GapDetectorResult {
-    DefaultGapDetectorResult,
-    ParquetFileGapDetectorResult,
+    DefaultGapDetectorResult(DefaultGapDetectorResult),
+    ParquetFileGapDetectorResult(ParquetFileGapDetectorResult),
```

Comment:
This is annoying that we have to use an enum like this to handle both types of processors... hopefully in SDK we can create a DefaultVersionTracker and a ParquetVersionTracker

Reply:
Agreed!! need to use so many enums to handle both types. Looking forward to cleaning this up using SDK !